### PR TITLE
parseConfigNodeLink support Chinese database name #2231

### PR DESCRIPTION
**Pull request by sanrentai**

### CODE REVIEW

File: database/gdb/gdb.go [hunk 1]
```diff
@@ -305,7 +305,7 @@ const (
 	ctxKeyInternalProducedSQL gctx.StrKey = `CtxKeyInternalProducedSQL`
 
 	// type:[username[:password]@][protocol[(address)]]/dbname[?param1=value1&...&paramN=valueN]
-	linkPattern = `(\w+):([\w\-]*):(.*?)@(\w+?)\((.+?)\)/{0,1}([\w\-]*)\?{0,1}(.*)`
+	linkPattern = `(\w+):([\w\-]*):(.*?)@(\w+?)\((.+?)\)/{0,1}([一-龥\w\+]*)\?{0,1}(.*)`
```
Comment (gqcn) [inline database/gdb/gdb.go:308]: How about `(\w+):([\w\-]*):(.*?)@(\w+?)\((.+?)\)/{0,1}(.*)\?{0,1}(.*)`?

**Review by gqcn — CHANGES_REQUESTED** [review]

File: database/gdb/gdb.go (hunk 1 above)
Comment (sanrentai) [reply database/gdb/gdb.go:308]:
```
func main() {
	linkPattern1 := `(\w+):([\w\-]*):(.*?)@(\w+?)\((.+?)\)/{0,1}([一-龥\w\+]*)\?{0,1}(.*)`
	linkPattern2 := `(\w+):([\w\-]*):(.*?)@(\w+?)\((.+?)\)/{0,1}(.*)\?{0,1}(.*)`
	link := `mysql:root:12345678@tcp(127.0.0.1:3306)/a正bc式?loc=Local&parseTime=true`

	m, err := gregex.MatchString(linkPattern1, link) 
	if err != nil {
		panic(err)
	}
	fmt.Println(m[6])
	// output a正bc式
	m2, err := gregex.MatchString(linkPattern2, link)
	if err != nil {
		panic(err)
	}
	fmt.Println(m2[6])
	// output a正bc式?loc=Local&parseTime=true
}
```

Comment (gqcn) [reply database/gdb/gdb.go:308]: Try this:
```go
package main

import (
	_ "github.com/gogf/gf/contrib/drivers/mysql/v2"
	"github.com/gogf/gf/v2/frame/g"
	"github.com/gogf/gf/v2/text/gregex"
)

func main() {
	linkPattern1 := `(\w+):([\w\-]*):(.*?)@(\w+?)\((.+?)\)/{0,1}([一-龥\w\+]*)\?{0,1}(.*)`
	linkPattern2 := `(\w+):([\w\-]*):(.*?)@(\w+?)\((.+?)\)/{0,1}([^\?]*)\?{0,1}(.*)`
	link := `mysql:root:12345678@tcp(127.0.0.1:3306)/a正bc式?loc=Local&parseTime=true`

	m1, err := gregex.MatchString(linkPattern1, link)
	if err != nil {
		panic(err)
	}
	g.Dump(m1)

	m2, err := gregex.MatchString(linkPattern2, link)
	if err != nil {
		panic(err)
	}
	g.Dump(m2)
}
```

Comment (sanrentai) [reply database/gdb/gdb.go]: linkPattern1 and linkPattern2 both get same result.

**Review by gqcn — APPROVED** [review]

**Pull request merged**